src/execution_plan/optimizations/compact_filters.c

```diff
@@ -37,7 +37,7 @@ static void _removeTrueFilter(ExecutionPlan *plan, OpBase *op) {
 		SIValue_Free(bool_val);
 		return;
 	}
-	if(SIValue_IsTrue(bool_val)) {
+	if(!SIValue_IsNull(bool_val) && SIValue_IsTrue(bool_val)) {
```

Comment:
At this point `bool_val` can only be either NULL or Boolean,
In case it is NULL, do we want to remove the filter ? I can't recall if a filter tree evaluated to NULL means the filter had passed or not?

Reply:
to pass